### PR TITLE
OLED Module C

### PR DESCRIPTION
**Pull request by leobeo-png**
Replaced the code for the oled display with C code. The usage time is still not integrated in it.

### CODE REVIEW

**Review by P1p1Kaka — APPROVED** [review]
Looks good to me :)

File: src/OLEDScreen.cpp [hunk 1]
```diff
@@ -1,358 +1,154 @@
-#include "OLEDScreen.h"
+#include <IoModule.h>
 
-// U8X8_SSD1306_128X64_NONAME_HW_I2C u8x8 (U8X8_PIN_NONE); //pin used A4 (SDA) and A5 (SCL)
-U8G2_SSD1306_128X64_NONAME_1_HW_I2C u8g2(U8G2_R0);// [page buffer, size = 128 B]
+#define SSD1306_ADDR  0x78
```
Comment (ThunFjsch) [inline src/OLEDScreen.cpp]: Is this a specific address you need? Maybe make clear what this address is for with a comment? As well as the choice for the hex value 0x78.

File: src/OLEDScreen.cpp [hunk 2]
```diff
@@ -1,358 +1,154 @@
...
+int state = 0;
+int speed = 0;
+bool debug = false;
 
-unsigned long lastSaveTime = 0; // Variable to store the last time data was saved
-const unsigned long saveInterval = 45000; // Save data every 45 seconds (in milliseconds)
+u8g2_t u8g2;
 
+/*
+enum time {
```
Comment (ThunFjsch) [inline src/OLEDScreen.cpp]: if it is commented out and not needed just delete it.

File: src/OLEDScreen.cpp [hunk 3]
```diff
@@ -1,358 +1,154 @@
...
-}
-
-void setForwBackDirection(bool newForwBackDirection) {
-  directionForwBack = newForwBackDirection;
-}
-
-void setLeftRightDirection(bool newLeftRightDirection){
-  directionLeftRight = newLeftRightDirection;
+void initIOModule(){
+	//u8g2_Setup_ssd1306_i2c_128x32_univision_f(&u8g2, U8G2_R0, u8x8_byte_avr_hw_i2caadasd, u8x8_avr_delay);
```
Comment (ThunFjsch) [inline src/OLEDScreen.cpp]: This comment can also be deleted

File: src/OLEDScreen.cpp [hunk 4]
```diff
@@ -1,358 +1,154 @@
...
+		u8g2_DrawStr(&u8g2, 73, 12, timeString);
+		*/
+		// Drawing the other non-animated elements
+		drawStaticElements();
+	}while (u8g2_NextPage(&u8g2));
 }
 
-void batteryStatus() { // read the battery voltage
-
+int main (void)
```
Comment (ThunFjsch) [inline src/OLEDScreen.cpp]: delete the main here.

File: src/OLEDScreen.h [hunk 5]
```diff
@@ -1,215 +1,213 @@
 #ifndef IOModule_H
 #define  IOModule_H
-#include <U8g2lib.h>
-#include <Arduino.h>
-#include <EEPROM.h>
+#include <u8g2.h>
+#include <u8x8_avr.h>
+#include <stdbool.h>
+#include <avr/io.h>
+#include <util/delay.h>
```
Comment (ThunFjsch) [inline src/OLEDScreen.h]: Do you need the util/delay?

File: src/OLEDScreen.h [hunk 6]
```diff
@@ -1,215 +1,213 @@
 #ifndef IOModule_H
 #define  IOModule_H
-#include <U8g2lib.h>
-#include <Arduino.h>
-#include <EEPROM.h>
+#include <u8g2.h>
+#include <u8x8_avr.h>
+#include <stdbool.h>
+#include <avr/io.h>
```
Comment (ThunFjsch) [inline src/OLEDScreen.h]: this is imported in the main.c

File: src/OLEDScreen.h [hunk 7]
```diff
@@ -1,215 +1,213 @@
 #ifndef IOModule_H
 #define  IOModule_H
-#include <U8g2lib.h>
-#include <Arduino.h>
-#include <EEPROM.h>
+#include <u8g2.h>
+#include <u8x8_avr.h>
+#include <stdbool.h>
```
Comment (ThunFjsch) [inline src/OLEDScreen.h]: this include can be applied in the main.c

**Review by ThunFjsch — CHANGES_REQUESTED** [review]

**Review by ThunFjsch — APPROVED** [review]

**Pull request merged**